src/metatrain/experimental/phace/tests/test_basic.py

```diff
@@ -0,0 +1,188 @@
+
+    def test_torchscript_integers(
+        self, model_hypers: dict, dataset_info: DatasetInfo
+    ) -> None:
+        """Tests that the model can be jitted when some float
+        parameters are instead supplied as integers.
+
+        :param model_hypers: Hyperparameters to initialize the model.
+        :param dataset_info: Dataset to initialize the model.
+        """
+
+        new_hypers = copy.deepcopy(model_hypers)
+        for hyper in self.float_hypers:
+            nested_key = hyper.split(".")
+            sub_dict = new_hypers
+            for key in nested_key[:-1]:
+                sub_dict = sub_dict[key]
+            sub_dict[nested_key[-1]] = int(sub_dict[nested_key[-1]])
+
+        model = self.model_cls(new_hypers, dataset_info)
+
+        system = System(
+            types=torch.tensor([6, 1, 8, 7]),
+            positions=torch.tensor(
+                [[0.0, 0.0, 0.0], [0.0, 0.0, 1.0], [0.0, 0.0, 2.0], [0.0, 0.0, 3.0]]
+            ),
+            cell=torch.zeros(3, 3),
+            pbc=torch.tensor([False, False, False]),
+        )
+        system = get_system_with_neighbor_lists(
+            system, model.requested_neighbor_lists()
+        )
+
+        model.module = model.fake_gradient_model
+        del model.gradient_model
+        del model.fake_gradient_model
+
+        model = torch.jit.script(model)
+        model(
+            [system],
+            model.outputs,
+        )
+
+
+class TestCheckpoints(CheckpointTests, PhACETests):
+    incompatible_trainer_checkpoints = []
+
+    @pytest.fixture
+    def model_trainer(
+        self,
+        dataset_path: str,
+        dataset_targets: dict,
+        minimal_model_hypers: dict,
+        default_hypers: dict,
+    ) -> tuple[ModelInterface, TrainerInterface]:
+        # Load dataset
+        dataset, targets_info, dataset_info = self.get_dataset(
+            dataset_targets, dataset_path
+        )
+
+        # Initialize model
+        model = self.model_cls(minimal_model_hypers, dataset_info)
+
+        # Set the training hyperparameters:
+        #  - Just 1 epoch to keep the test fast
+        #  - Default loss for each target
+        hypers = copy.deepcopy(default_hypers)
+        hypers["training"]["compile"] = False
+        hypers["training"]["num_epochs"] = 1
+        loss_hypers = OmegaConf.create(
+            {k: init_with_defaults(LossSpecification) for k in dataset_targets}
+        )
+        loss_hypers = OmegaConf.to_container(loss_hypers, resolve=True)
+        hypers["training"]["loss"] = loss_hypers
+
+        # Initialize trainer
+        trainer = self.trainer_cls(hypers["training"])
+
+        # Train the model.
+        trainer.train(
+            model,
+            dtype=model.__supported_dtypes__[0],
+            devices=[torch.device("cpu")],
+            train_datasets=[dataset],
+            val_datasets=[dataset],
+            checkpoint_dir="",
+        )
+
+        return model, trainer
```

Comment:
For this you can do 

```python
class TestCheckpoints:
    @pytest.fixture
    def default_hypers(self):
        """Train without compiling"""
        hypers = get_default_hypers(self.architecture)
        hypers = copy.deepcopy(hypers)
        hypers["training"]["compile"] = False
        return hypers
```

Or even put this fixture in the `PhACETests` class unless you ever want to train with `compile` in the tests. Is this the reason why you also removed the `TestTraining` class, or this is for some other reason?

Reply:
Hmm I don't see why not, the only thing you seem to change in the `model_trainer` fixture is the `compile` hyper.

Anyway, there is no overspecialization at all, you are free to write any tests that you want if you think the provided ones are not suitable for your architecture. I just think it's nice to use the experience of each new architecture to see if we can abstract things away to make it easier for the next person.

Reply:
You're welcome to push it if you can make it work. I couldn't